saas/sdk/app/auth.py

```diff
@@ -271,13 +310,28 @@ def verify_password(cls, plain_password: str, hashed_password: str) -> bool:
     @classmethod
     def _authenticate_user(cls, login: str, password: str) -> Optional[User]:
         user = UserDB.get_user(login)
+        # not a user
         if not user:
             return None
 
-        if not cls.verify_password(password, user.hashed_password):
-            return None
+        # user is not disabled? verify password
+        if not user.disabled:
+            if not cls.verify_password(password, user.hashed_password):
+                # if password is incorrect, update the failed login attempts count
+                user = UserDB.update_login_attempts(login, False)
+                # if the login attempts count has reached the maximum value, disable the user
+                if user.login_attempts >= cls._max_login_attempts:
+                    UserDB.disable_user(login)
+                    return 'locked'
```

Comment:
A string is returned but the signature of the function defines an (optional) User object to be returned.